core/src/main/scala/cats/data/Newtype.scala

```diff
@@ -0,0 +1,12 @@
+package cats
+package data
+
+/**
+  * Helper trait for `newtype`s. These allow you to create a zero-allocation wrapper around a specific type.
+  * Similar to `AnyVal` value classes, but never have any runtime overhead.
+  */
+trait Newtype { self =>
```

Comment:
Does it make sense for this to be a public trait if `Base` and `Tag` are both `private[data]`?

Reply:
Debatable IMO, but to me those two are implementation details, you're never actually going to want to access these, no?

Reply:
Oh right I guess that in order to create your own type that uses `Newtype` you don't actually have to set `Base` and `Tag`, right?

I guess I'm partially hesitant to expose `Newtype` publicly because there are a lot of approaches to newtypes in the scala world and I'm not sure whether we want to take on the burden of keeping this one around and stable in cats. Especially since it's an internal implementation detail that isn't really used as an abstraction and is trivial enough that people could easily copy/paste it if they wanted it. I'd be inclined to make it private for now and if it works out well and people want it to be exposed publicly it's easier to open it up later than to make it private later. Admittedly I'm sometimes overly paranoid about this sort of thing...

Reply:
I'd be okay with making them private :)